src/StreamFactory/DiactorosStreamFactory.php

```diff
@@ -18,26 +18,19 @@
      */
     public function createStream($body = null)
     {
-        if (!$body instanceof StreamInterface) {
-            if (is_resource($body)) {
-                $body = new Stream($body);
-            } else {
-                $stream = new Stream('php://memory', 'rw');
-
-                if (null === $body || '' === $body) {
-                    return $stream;
-                }
-
-                $stream->write((string) $body);
+        if ($body instanceof StreamInterface) {
+            return $body;
+        }
 
-                $body = $stream;
-            }
+        if (is_resource($body)) {
+            return new Stream($body);
         }
 
-        if ($body->isSeekable()) {
-            $body->rewind();
```

Comment:
Since we do not have to rewind the stream I could reduce the complexity and increase readability of the factories.